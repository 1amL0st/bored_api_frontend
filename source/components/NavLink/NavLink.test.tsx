import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { NavLink } from './index';

describe('NavLink component tests', () => {
  test('Plain text as child', () => {
    const component = renderer.create(
      <BrowserRouter>
        <NavLink to="somewhere">Button</NavLink>
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Div as child', () => {
    const component = renderer.create(
      <BrowserRouter>
        <NavLink to="somewhere">
          <div>Hello</div>
        </NavLink>
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
