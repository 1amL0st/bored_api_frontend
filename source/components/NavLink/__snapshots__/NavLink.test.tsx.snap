// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Button test 1`] = `
<a
  aria-current={null}
  className="nav-link"
  href="/somewhere"
  onClick={[Function]}
>
  Button
</a>
`;

exports[`My Test Button with child div 1`] = `
<a
  aria-current={null}
  className="nav-link"
  href="/somewhere"
  onClick={[Function]}
>
  <div>
    Hello
  </div>
</a>
`;

exports[`NavLink component tests Div as child 1`] = `
<a
  aria-current={null}
  className="nav-link"
  href="/somewhere"
  onClick={[Function]}
>
  <div>
    Hello
  </div>
</a>
`;

exports[`NavLink component tests Plain text as child 1`] = `
<a
  aria-current={null}
  className="nav-link"
  href="/somewhere"
  onClick={[Function]}
>
  Button
</a>
`;
